src/ProductDetails.js

```diff
@@ -0,0 +1,104 @@
+import styled from "styled-components";
+import { useParams } from "react-router-dom";
+import { useProductById } from "./utils/hooks/useProductById";
+import { useProductCategories } from "./utils/hooks/useProductCategories";
+import Slideshow from "./Slider/Slideshow";
+
+const ShopcartButton = styled.button`
+  display: block;
+  margin-left: auto;
+  margin-right: auto;
+  padding: 5px 20px 5px 20px;
+  background-color: rgba(0, 150, 255, 0.8);
+  font-size: 20px;
+  font-weight: bold;
+  text-decoration: none;
+  color: black;
+  border-radius: 15px;
+
+  &:hover {
+    background-color: rgba(0, 150, 255, 0.2);
+  }
+`;
+
+const ProductLabel = styled.label`
+  display: block;
+`;
+
+const ProductDetails = () => {
+  const params = useParams();
+
+  const { data: categoriesData, isLoading: isLoadingCategories } =
+    useProductCategories();
+
+  const {
+    data: { results },
+    isLoading: isLoadingProduct,
+  } = useProductById(params.productId);
+
+  let productSlides = [];
+  let product = null;
+
+  let category = "";
+
+  if (isLoadingProduct === false) {
```

Comment:
Esta línea se puede cambiar a algo más sencillo como 
```suggestion
  if ( !isLoadingProduct ) {
```
Solo usamos la negación,  porque no va a tener más valores distintos a `true` o `false`.